toad-cache.d.cts

```diff
@@ -0,0 +1,123 @@
+
+interface ToadCache<T> {
+    first: any;
+    last: any;
+    max: number;
+    ttl: number;
+    size: number;
+    clear(): void;
+    delete(key: any): void;
+    deleteMany(keys: any[]): void;
+    evict(): void;
+    expiresAt(key: any): any;
+    keys(): any;
+    get(key: any): T | undefined;
+    getMany(keys: any[]): (T | undefined)[];
+    set(key: any, value: T): void;
+}
+
+declare class FifoMap<T> implements ToadCache<T>{
+    constructor(max?: number, ttlInMsecs?: number);
+    first: any;
+    items: Map<any, any>;
+    last: any;
+    max: number;
+    ttl: number;
+    size: number;
+    clear(): void;
+    delete(key: any): void;
+    deleteMany(keys: any[]): void;
+    evict(): void;
+    expiresAt(key: any): any;
+    get(key: any): T | undefined;
+    getMany(keys: any[]): (T | undefined)[];
+    keys(): IterableIterator<any>;
+    set(key: any, value: T): void;
+}
+
+declare class FifoObject<T> implements ToadCache<T> {
+    constructor(max?: number, ttlInMsecs?: number);
+    first: any;
+    items: any;
+    last: any;
+    size: number;
+    max: number;
+    ttl: number;
+    clear(): void;
+    delete(key: any): void;
+    deleteMany(keys: any[]): void;
+    evict(): void;
+    expiresAt(key: any): any;
+    get(key: any): T | undefined;
+    getMany(keys: any[]): (T | undefined)[];
+    keys(): string[];
+    set(key: any, value: T): void;
+}
+
+declare class LruMap<T> implements ToadCache<T> {
+    constructor(max?: number, ttlInMsecs?: number);
+    first: any;
+    items: Map<any, any>;
+    last: any;
+    max: number;
+    ttl: number;
+    size: number;
+    clear(): void;
+    delete(key: any): void;
+    deleteMany(keys: any[]): void;
+    evict(): void;
+    expiresAt(key: any): any;
+    get(key: any): T | undefined;
+    getMany(keys: any[]): (T | undefined)[];
+    keys(): IterableIterator<any>;
+    set(key: any, value: T): void;
+}
+
+declare class LruObject<T> implements ToadCache<T> {
+    constructor(max?: number, ttlInMsecs?: number);
+    first: any;
+    items: any;
+    last: any;
+    size: number;
+    max: number;
+    ttl: number;
+
+    clear(): void;
+    delete(key: any): void;
+    deleteMany(keys: any[]): void;
+    evict(): void;
+    expiresAt(key: any): any;
+    get(key: any): T | undefined;
+    getMany(keys: any[]): (T | undefined)[];
+    keys(): string[];
+    set(key: any, value: T): void;
+}
+
+declare class HitStatisticsRecord {
+    records: Record<string, Record<string, {
+        expirations: number,
+        hits: number,
+        misses: number,
+    }>>
+
+    initForCache(cacheId: string, currentTimeStamp: string): void
+    resetForCache(cacheId: string): void
+}
+
+declare class LruObjectHitStatistics<T> extends LruObject<T>{
+    constructor(max?: number, ttlInMsecs?: number, cacheId?: string, globalStatisticsRecord?: HitStatisticsRecord, statisticTtlInHours?: number);
+}
+
+export {
```

Comment:
not sure about this syntax, is this the right CommonJS types way?..

Reply:
see https://github.com/arethetypeswrong/arethetypeswrong.github.io/blob/main/docs/problems/FalseESM.md for what this is trying to solve

Reply:
why not doing it like we do it in fastify?

Reply:
fastify is CJS only: https://arethetypeswrong.github.io/?p=fastify%404.24.3